app/src/main/java/com/infomaniak/mail/views/DecoratedTextItemView.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Infomaniak ikMail - Android
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.views
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.annotation.StringRes
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.content.res.getDimensionPixelSizeOrThrow
+import androidx.core.view.isGone
+import com.google.android.material.shape.ShapeAppearanceModel
+import com.infomaniak.lib.core.utils.getAttributes
+import com.infomaniak.lib.core.utils.setMargins
+import com.infomaniak.lib.core.utils.setMarginsRelative
+import com.infomaniak.mail.R
+import com.infomaniak.mail.databinding.ViewDecoratedTextItemBinding
+import com.infomaniak.lib.core.R as RCore
+
+abstract class DecoratedTextItemView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    val binding by lazy { ViewDecoratedTextItemBinding.inflate(LayoutInflater.from(context), this, true) }
+
+    open val endIconMarginRes: Int = RCore.dimen.marginStandardVerySmall
```

Comment:
Currently we're either at 20dp or 24dp for icons instead of being either at 16dp or 24dp from the very edge of the item, aren't we?

Reply:
No, it's centered-aligned because pastille is 8dp(size) + 8dp(margin) so 12 from the center of the icon, and other icons are 16 (size) + 4(margin), so 12dp between center and end too